src/restartable.js

```diff
@@ -169,7 +169,11 @@ export default function restartable (driver, opts = {}) {
       }
     }
 
-    lastSinkEvent$.imitate(sink$);
+    if (!(sink$ instanceof MemoryStream)) {
```

Comment:
I don't think this needs to be conditional, we can always do `filter(() => {})` and it will be fine.

It probably justifies a comment though, explaining that this is necessary because sometimes the `sink$` might be a `MemoryStream`.